### PR TITLE
Field trip change travel date

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
Requires/blocked by #620.
Fix https://github.com/ibi-group/trimet-mod-otp/issues/307
Also contains a refactor for the code that updates attributes on a field trip.

For easier review, look at the changes before the one about lint.

### CODE REVIEW

File: lib/components/admin/field-trip-details.js [hunk 1]
```diff
@@ -73,9 +82,16 @@ class FieldTripDetails extends Component {
...
-    const convertedRequestDate = moment(evt.target.value).format(dateFormat)
-    setRequestDate(request, convertedRequestDate, intl)
+    const newDate = evt.target.value
+    // Only persist the date if it is valid (not an empty string)
+    if (newDate !== '') {
+      const { dateFormat, intl, request, setRequestDate } = this.props
+      const convertedRequestDate = moment(newDate).format(dateFormat)
+      setRequestDate(request, convertedRequestDate, intl)
+      this.setState({
+        travelDateUpdateCount: this.state.travelDateUpdateCount + 1
```
Comment (miles-grant-ibigroup) [inline lib/components/admin/field-trip-details.js:92]: Do we use this count for anything else than updating the key? If not, would using `uuid()` not make more sense? This is what we do elsewhere when we have this problem -- the library is already in the `yarn.lock`

File: lib/actions/field-trip.js [hunk 2]
```diff
@@ -122,22 +206,23 @@ export function fetchFieldTripDetails (requestId, intl) {
...
     const noteData = serialize({
-      note: {...note, userName: username},
+      note: { ...note, userName: username },
       requestId: request.id,
       sessionId
     })
-    return fetch(`${datastoreUrl}/fieldtrip/addNote`,
-      {body: noteData, method: 'POST'}
-    )
+    return fetch(`${datastoreUrl}/fieldtrip/addNote`, {
```
Comment (miles-grant-ibigroup) [inline lib/actions/field-trip.js:220]: Is this fetch not able to use the new request method?

File: lib/actions/field-trip.js [hunk 3]
```diff
@@ -149,192 +234,114 @@ export function addFieldTripNote (request, note, intl) {
...
-            { id: 'actions.fieldTrip.editSubmitterNotesError' },
-            { err: JSON.stringify(err) }
-          )
-        )
-      })
-  }
+export function deleteFieldTripNote(request, noteId, intl) {
+  return updateFieldTripRequest(
+    request,
+    'deleteNote',
```
Comment (miles-grant-ibigroup) [inline lib/actions/field-trip.js:274]: Do you think it might be a good idea to make these an enum to avoid room for error when editing this in the future

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Don't have much experience with the calltaker module, but changing the date seems to work and the linted code looks great! Thanks for handling that linting these huge files is so annoying.

File: lib/actions/field-trip.js (hunk 3 above)
Comment (binh-dam-ibigroup) [reply lib/actions/field-trip.js]: Handled in https://github.com/opentripplanner/otp-react-redux/pull/627/commits/42a8fb2c0864378234354ceb9827ddf380ac03bc.

File: lib/actions/field-trip.js (hunk 2 above)
Comment (binh-dam-ibigroup) [reply lib/actions/field-trip.js]: Handled in https://github.com/opentripplanner/otp-react-redux/pull/627/commits/42a8fb2c0864378234354ceb9827ddf380ac03bc.

**Comment by binh-dam-ibigroup** [comment]
> 
It did when clearing one of the date fields then selecting a different field trip with the same original travel date! I fixed that in https://github.com/opentripplanner/otp-react-redux/pull/627/commits/cf6df2005fe5aebcc8324738f7e5c56ddde4976f.

**Review by miles-grant-ibigroup — APPROVED** [review]
Thanks for the changes! Still works great. 

Once the merge conflicts are resolved this is ready to go!

**Comment by demory** [comment]
I'm getting merge conflicts too when I try to test this -- what is the status of those?

**Comment by binh-dam-ibigroup** [comment]
> I'm getting merge conflicts too when I try to test this -- what is the status of those?

Github says the PR is mergeable... I'll take a look at what you have.

**Review by demory — APPROVED** [review]

**Pull request merged**